src/common/transformations/src/transformations/common_optimizations/fuse_rotary_positional_embeddings.cpp

```diff
@@ -542,6 +542,134 @@ ov::pass::RoPEFusionChatGLM::RoPEFusionChatGLM(int split_output_id) {
     this->register_matcher(m, callback);
 }
 
+ov::pass::RoPEFusionChatGLM4::RoPEFusionChatGLM4(int split_output_id) {
+    MATCHER_SCOPE(RoPEFusionChatGLM4);
+
+    auto qkv_linear = makePattern("[?,?,?]");  //  [seq_length, batch_size, 4608]
+    auto seq_length = makePattern("i32[1]");
+    auto cos_sin_cache = makePattern("[?,?,?,?]");  // [max_pos_embeddings, batch_size, 32, 2]
+
+    auto ndims = ov::gen_pattern::Symbol("ndims");
+    auto head_cnt = ov::gen_pattern::Symbol("head_cnt");
+    auto head_size = ov::gen_pattern::Symbol("head_size");
+    auto total_size_q = ov::gen_pattern::Symbol("total_size_q");
+    auto total_size_k = ov::gen_pattern::Symbol("total_size_k");
+    auto total_size_v = ov::gen_pattern::Symbol("total_size_v");
+    auto batch = ov::gen_pattern::Symbol("batch");
+    auto seq_len = ov::gen_pattern::Symbol("seq_len");
+
+    auto qkv_proj = makePattern<opset1::VariadicSplit>({qkv_linear, -1, {total_size_q, total_size_k, total_size_v}});
+    qkv_proj->set_output_size(3);
+
+    // get key [L, B, Hkv, S]
+    auto cur_key = makePattern<opset1::Reshape>({qkv_proj->output(split_output_id), {0, 0, head_cnt, head_size}},
+                                                {{"special_zero", true}});
+
+    auto cur_key_transposed = makePattern<opset1::Transpose>({cur_key, {0, 2, 1, 3}});
```

Comment:
The transpose of key with axes `{0,2,1,3}` could be described with a common option `config.input_trans0213 = true`. I think it's better to use general option so that the kernel development could be more general.